src/test/java/jewellery/inventory/unit/service/SaleServiceTest.java

```diff
@@ -0,0 +1,122 @@
+  @Mock private ProductRepository productRepository;
+  @Mock private SaleMapper saleMapper;
+  private User seller;
+  private User buyer;
+  private Product product;
+  private Sale sale;
+  private SaleRequestDto saleRequestDto;
+  private SaleRequestDto saleRequestDtoOwnerEqualsRecipient;
+  private SaleResponseDto saleResponseDto;
+  private ProductPriceDiscountRequestDto productPriceDiscountRequestDto;
+  private List<SaleResponseDto> saleResponseDtoList;
+  private List<Product> productsForSale;
+
+  @BeforeEach
+  void setUp() {
+    seller = createTestUserForSale();
+    buyer = createSecondTestUser();
+    product = getTestProduct(seller, new Resource());
+    productsForSale = SaleTestHelper.getProList(product);
+    sale = SaleTestHelper.createSaleWithTodayDate(seller, buyer, productsForSale);
+    saleResponseDto = SaleTestHelper.getSaleResponseDto(sale);
+    productPriceDiscountRequestDto =
+        SaleTestHelper.createProductPriceDiscountRequest(product.getId(), 1000, 10);
+    List<ProductPriceDiscountRequestDto> productPriceDiscountRequestDtoList = new ArrayList<>();
+    productPriceDiscountRequestDtoList.add(productPriceDiscountRequestDto);
+    saleRequestDto =
+        SaleTestHelper.createSaleRequest(
+            seller.getId(), buyer.getId(), productPriceDiscountRequestDtoList);
+    saleRequestDtoOwnerEqualsRecipient =
+            SaleTestHelper.createSaleRequest(
+                    seller.getId(), seller.getId(), productPriceDiscountRequestDtoList);
+    saleResponseDtoList = SaleTestHelper.getSaleResponseList(saleResponseDto);
+  }
+
+  @Test
+  void testGetAllSales() {
+
+    List<Sale> sales = Arrays.asList(sale, new Sale(), new Sale());
+
+    when(saleRepository.findAll()).thenReturn(sales);
+
+    List<SaleResponseDto> responses = saleService.getAllSales();
+
+    Assertions.assertEquals(sales.size(), responses.size());
+  }
+
+  @Test
+  void testCreateSaleProductWhenDateIsCorrect() {
+    when(saleMapper.mapRequestToEntity(saleRequestDto)).thenReturn(sale);
+
+    when(saleRepository.save(sale)).thenReturn(sale);
+
+    when(saleMapper.mapEntityToResponseDto(sale)).thenReturn(saleResponseDto);
+
+    SaleResponseDto actual = saleService.createSale(saleRequestDto);
+    assertNotEquals(actual.getBuyer(), actual.getSeller());
+    Assertions.assertEquals(saleRequestDto.getSellerId(), actual.getSeller().getId());
+    assertNotNull(actual);
+  }
+
+  @Test
+  void testCreateSaleProductWhenProductOwnerEqualsRecipientException() {
+    when(saleMapper.mapRequestToEntity(saleRequestDtoOwnerEqualsRecipient)).thenReturn(sale);
+
+    assertThrows(
+            ProductOwnerEqualsRecipientException.class,
+            () -> saleService.createSale(saleRequestDtoOwnerEqualsRecipient));
+  }
+
+  @Test
+  void testProductOwnerNotSeller() {
+    UUID ownerId = UUID.randomUUID();
+    UUID sellerId = UUID.randomUUID();
+
+    try {
+      throw new ProductOwnerNotSeller(ownerId, sellerId);
```

Comment:
We are supposed to stub the repository (or another related dependency) so that the service create method throws the exception. Right now we are just testing the constructor of the ProductOwnerNotSeller exception.